tokio/src/io/util/buf_reader.rs

```diff
@@ -45,7 +46,12 @@ impl<R: AsyncRead> BufReader<R> {
         unsafe {
             let mut buffer = Vec::with_capacity(capacity);
             buffer.set_len(capacity);
-            inner.prepare_uninitialized_buffer(&mut buffer);
+
+            {
+                // Convert to MaybeUninit
+                let b = &mut *(&mut buffer[..] as *mut [u8] as *mut [MaybeUninit<u8>]);
```

Comment:
TIOLI: this shows up in a couple places, should there be a util fn for it or something?

Reply:
I'll leave for now as we're hoping this entire section will be updated when we figure out how to deal w/ uninitialized mem & reading :)